Source/Editor/DualityEditor/Utility/SerializableReferenceWrapper.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.Serialization;
+using Duality.Serialization;
+
+namespace Duality.Editor
+{
+	/// <summary>
+	/// A wrapper object that stores a reference to a non-<see cref="SerializableAttribute"/> object.
+	/// </summary>
+	[Serializable]
+	public class SerializableReferenceWrapper : SerializableWrapper
+	{
+		private static long nextID = 0;
+		private static readonly Dictionary<long, object> referenceMap 
+			= new Dictionary<long, object>();
+
+		private long ID = -1;
+
+		public sealed override object Data
+		{
+			get
+			{
+				return this.ID < 0 
+					? null 
+					: referenceMap[this.ID];
+			}
+			set
+			{
+				if (this.ID < 0)
+					this.ID = nextID++;
+
+				referenceMap[this.ID] = value;
```

Comment:
We shouldn't set an ID and store a reference item immediately when setting Data, but only when actually serializing it for the first time. That way we avoid cluttering the reference map with items for every `DataObject` out there that might store wrapped data.